File: 53-MaximumSubarray/53-MaximumSubarray.java
// Last updated: 8/11/2026, 12:26:26 PM
class Solution {
    public int maxSubArray(int[] nums) {
        int maxSum = nums[0], currentSum = nums[0];
        for (int i = 1; i < nums.length; i++) {
            currentSum = Math.max(nums[i], currentSum + nums[i]);
            maxSum = Math.max(maxSum, currentSum);
        }
        return maxSum;
    }
}

File: 53-MaximumSubarray/53-MaximumSubarray.py
# Last updated: 9/2/2026, 12:44:26 PM
class Solution:
    def maxSubArray(self, nums):
        current = nums[0]
        answer = nums[0]

        for num in nums[1:]:
            current = max(num, current + num)
            answer = max(answer, current)

        return answer